### PR TITLE
Add fluentd-plugin-kafka to fluentd image

### PR DESCRIPTION
**Pull request by periklis**
This PR vendors the [fluentd-plugin-kafka](https://github.com/fluent/fluent-plugin-kafka) into the fluentd image.

### CODE REVIEW

File: fluentd/Gemfile [hunk 1]
```diff
@@ -5,6 +5,7 @@ gem 'elasticsearch'
 gem 'fluent-plugin-kubernetes_metadata_filter'
 gem 'fluent-plugin-concat'
 gem 'fluent-plugin-elasticsearch'
+gem 'fluent-plugin-kafka'
```
Comment (jcantrill) [inline fluentd/Gemfile]: @alanconway do you think we should start locking down versions to be repeatable here and not be "surprised" by updates.

Comment (periklis) [reply fluentd/Gemfile]: @jcantrill For this purpose i've added 767c29f5b2f2f05982884da73d3be50dd25e98fa to expedite this PR.

File: fluentd/Gemfile [hunk 2]
```diff
@@ -5,6 +5,7 @@ gem 'elasticsearch'
 gem 'fluent-plugin-kubernetes_metadata_filter'
 gem 'fluent-plugin-concat'
 gem 'fluent-plugin-elasticsearch'
+gem 'fluent-plugin-kafka', ENV['FLUENTD_PLUGIN_KAFKA_VERSION']
```
Comment (jcantrill) [inline fluentd/Gemfile]: I might recommend the opposite change here and set the version in the Gemfile and not Dockerfile.  I don't see the advantage of it existing in the Dockerfile at all for this or fluentd unless there is some dependency there.

**Comment by periklis** [comment]
@jcantrill I've removed the extra ENV check for `fluentd-plugin-kafka`. In addition i've split the PR in three more obvious commits to enable readability.

**Comment by periklis** [comment]
Please ignore 109d3ce for. Depends on: #1858

**Pull request merged**